extensions/gdx-setup/src/com/badlogic/gdx/setup/GdxSetup.java

```diff
@@ -253,17 +253,30 @@ public void build (ProjectBuilder builder, String outputDir, String appName, Str
 		project.files.add(new ProjectFile("gradle/wrapper/gradle-wrapper.properties", false));
 		project.files.add(new ProjectFile("gradle.properties"));
 
-		// core project
+		// CoreGdxDefinition project
 		project.files.add(new ProjectFile("core/build.gradle"));
-		project.files.add(new ProjectFile("core/src/MainClass", "core/src/" + packageDir + "/" + mainClass + ".java", true));
+
+		if (language == Language.JAVA) {

```

Comment:
Ok, i'm done with abstracting resource out, and also fixed a problem regarding to running ios-moe in separate commit. Please check them out. Thanks!